python/lsst/daf/butler/registry/datasets/byDimensions/_storage.py

```diff
@@ -549,119 +548,6 @@ def getDataId(self, id: DatasetId) -> DataCoordinate:
-            datasetId = dataset.id
-            if datasetId is None:
-                # if reuseIds is set then all IDs must be known
-                if reuseIds:
-                    raise TypeError("All dataset IDs must be known if `reuseIds` is set")
-            elif isinstance(datasetId, int):
-                if reuseIds:
-                    datasetIdList.append(datasetId)
-            else:
-                raise TypeError(f"Unsupported type of dataset ID: {type(datasetId)}")
-
-        yield from self._insert(run, dataIdList, datasetIdList)
-
-    def _insert(
-        self, run: RunRecord, dataIdList: list[DataCoordinate], datasetIdList: list[int] | None = None
-    ) -> Iterator[DatasetRef]:
-        """Common part of implementation of `insert` and `import_` methods."""
-
-        # Remember any governor dimension values we see.
-        summary = CollectionSummary()
-        summary.add_data_ids(self.datasetType, dataIdList)
-
-        staticRow = {
-            "dataset_type_id": self._dataset_type_id,
-            self._runKeyColumn: run.key,
-        }
-        with self._db.transaction():
-            # Insert into the static dataset table, generating autoincrement
-            # dataset_id values.
-            if datasetIdList:
-                # reuse existing IDs
-                rows = [dict(staticRow, id=datasetId) for datasetId in datasetIdList]
-                self._db.insert(self._static.dataset, *rows)
-            else:
-                # use auto-incremented IDs
-                datasetIdList = self._db.insert(
-                    self._static.dataset, *([staticRow] * len(dataIdList)), returnIds=True
-                )
-                assert datasetIdList is not None
-            # Update the summary tables for this collection in case this is the
-            # first time this dataset type or these governor values will be
-            # inserted there.
-            self._summaries.update(run, [self._dataset_type_id], summary)
-            # Combine the generated dataset_id values and data ID fields to
-            # form rows to be inserted into the tags table.
-            protoTagsRow = {
-                "dataset_type_id": self._dataset_type_id,
-                self._collections.getCollectionForeignKeyName(): run.key,
-            }
-            tagsRows = [
-                dict(protoTagsRow, dataset_id=dataset_id, **dataId.byName())
-                for dataId, dataset_id in zip(dataIdList, datasetIdList)
-            ]
-            # Insert those rows into the tags table.  This is where we'll
-            # get any unique constraint violations.
-            self._db.insert(self._tags, *tagsRows)
-
-        for dataId, datasetId in zip(dataIdList, datasetIdList):
-            yield DatasetRef(
-                datasetType=self.datasetType,
-                dataId=dataId,
-                id=datasetId,
-                run=run.name,
-            )
-
-
```

Comment:
It'd be a nice simplification to drop the intermediate base class and merge it with the remaining derived class, but I'm not sure that's worth holding up the ticket.